crates/cranelift/src/func_environ.rs

```diff
@@ -33,15 +33,14 @@ use wasmtime_environ::{FUNCREF_INIT_BIT, FUNCREF_MASK};
 pub(crate) struct BuiltinFunctions {
     types: BuiltinFunctionSignatures,
 
-    builtins:
-        [Option<ir::FuncRef>; BuiltinFunctionIndex::builtin_functions_total_number() as usize],
+    builtins: [Option<ir::FuncRef>; BuiltinFunctionIndex::max() as usize],
 }
 
 impl BuiltinFunctions {
     fn new(compiler: &Compiler) -> Self {
         Self {
             types: BuiltinFunctionSignatures::new(compiler),
-            builtins: [None; BuiltinFunctionIndex::builtin_functions_total_number() as usize],
+            builtins: [None; BuiltinFunctionIndex::max() as usize],
```

Comment:
Nitpick: If `max` returns the maximum index then this should be `max + 1`. If `max` returns the number of builtins then we should rename it to `len` or `count` or something.

Ditto just above.